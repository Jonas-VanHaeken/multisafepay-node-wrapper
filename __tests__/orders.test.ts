import axios from 'axios';
import Orders from '../src/resources/orders';
import createOrderMock from '../__mocks__/createOrder.mock.json';
import updateOrderMock from '../__mocks__/updateOrder.mock.json';
import createOrderResponse from '../__stubs__/createOrder.stub.json';
import MockAdapter from 'axios-mock-adapter';

const mock = new MockAdapter(axios);

describe('orders', () => {
  let orders: any;
  const error = { error_code: 1006, error_info: 'Invalid transaction ID' };
  beforeEach(() => {
    orders = new Orders(axios.create());
  });

  it('check endpoint', () => {
    expect(orders.endpoint).toBe('orders');
  });

  it('create order', async () => {
    mock.onPost('/orders', createOrderMock).reply(200, createOrderResponse);

    const resp = await orders.create(createOrderMock);

    expect(resp).toMatchSnapshot();
  });

  it('create order with error', async () => {
    const copyOrder = Object.assign({}, createOrderMock);
    copyOrder.order_id = 'my-order-id-1';

    mock.onPost('/orders', copyOrder).reply(404, error);

    const resp = await orders.create(copyOrder);

    expect(resp).toEqual(error);
  });

  it('update order', async () => {
    const mockResponse = {
      success: true,
      data: {},
    };
    mock
      .onPatch(`/orders/my-order-id-1a`, updateOrderMock)
      .reply(200, mockResponse);

    const resp = await orders.update('my-order-id-1a', updateOrderMock);

    expect(resp).toMatchSnapshot();
  });

  it('update order with error', async () => {
    mock
      .onPatch(`/orders/${createOrderMock.order_id}`, updateOrderMock)
      .reply(404, error);

    const resp = await orders.update(createOrderMock.order_id, updateOrderMock);

    expect(resp).toEqual(error);
  });
});
