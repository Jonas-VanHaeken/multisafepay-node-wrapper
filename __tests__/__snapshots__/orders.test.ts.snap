// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`orders create order 1`] = `
Object {
  "data": Object {
    "order_id": "my-order-id-1c",
    "payment_url": "https://testpayv2.multisafepay.com/connect/99iKkjUhWwGD03IgvxurNMUgVDYDPNpghWx/?lang=nl_NL",
  },
  "success": true,
}
`;

exports[`orders update order 1`] = `
Object {
  "data": Object {},
  "success": true,
}
`;
